flink/src/main/java/org/apache/iceberg/flink/FlinkCatalogFactory.java

```diff
@@ -101,11 +108,28 @@ static CatalogLoader createCatalogLoader(String name, Map<String, String> proper
     }
   }
 
+  @Override
+  public String factoryIdentifier() {
+    return IDENTIFIER;
+  }
+
+  @Override
+  public Set<ConfigOption<?>> requiredOptions() {
+    return Sets.newHashSet();
+  }
+
+  @Override
+  public Set<ConfigOption<?>> optionalOptions() {
+    final Set<ConfigOption<?>> options = Sets.newHashSet();
+    options.add(FactoryUtil.PROPERTY_VERSION);
+    return options;
+  }
+
   @Override
   public Map<String, String> requiredContext() {
```

Comment:
As the `requiredContext` and `supportedProperties` have been marked as `deprecated`,  I will suggest to implement those in `requiredOptions` & `optionalOptions`.

Reply:
As [`Context`](https://github.com/apache/flink/blob/5f007ff6c8224c6e5a14de2c79296eb85a22fe1f/flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/CatalogFactory.java#L66) class is missing in 1.12, I think we can only implement these deprecated methods.